test/sql/decimal_functions_sql_test.cpp

```diff
@@ -76,6 +114,221 @@ class DecimalFunctionsSQLTest : public PelotonTest {};
+    // Adding in 200 random decimal inputs between [-127, 127]
+    int i;
+    std::vector<int8_t> inputs;
+    int lo = 0;
+    int hi = 254;
+    int numEntries = 200;
+    // Setting a seed
+    std::srand(std::time(0));
+    for (i = 0; i < numEntries; i++) {
+      int32_t num = (std::rand() % (hi - lo)) - 127;
+      inputs.push_back(num);
+      std::ostringstream os;
+      os << "insert into foo values(" << i << ", " << num << ");";
+      TestingSQLUtil::ExecuteSQLQuery(os.str());
+    }
+    EXPECT_EQ(i, numEntries);
+    txn_manager.CommitTransaction(txn);
+
+    // Fetch values from the table
+    std::vector<ResultValue> result;
+    std::vector<FieldInfo> tuple_descriptor;
+    std::string error_message;
+    int rows_affected;
+    std::string sql_line = "select id, abs(income) from foo;";
+
+    TestingSQLUtil::ExecuteSQLQuery(sql_line, result, tuple_descriptor,
+                                    rows_affected, error_message);
+    for (i = 0; i < numEntries; i++) {
+      std::string result_id(
+              TestingSQLUtil::GetResultValueAsString(result, (2 * i)));
+      std::string result_income(
+              TestingSQLUtil::GetResultValueAsString(result, (2 * i) + 1));
+      int id = std::stoi(result_id);
+      int income = std::stoi(result_income);
+      EXPECT_EQ(id, i);
+      EXPECT_EQ(income, abs(inputs[i]));
+    }
+  }
+
+  TEST_F(DecimalSQLTestsBase, SmallIntAbsTest) {
+    CreateTableWithCol("smallint");
+    auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+    auto txn = txn_manager.BeginTransaction();
+    // Adding in 200 random integer inputs between [-32767, 32767]
+    int i;
+    std::vector<int16_t> inputs;
+    int lo = 0;
+    int hi = 65534;
+    int numEntries = 200;
+    // Setting a seed
+    std::srand(std::time(0));
+    for (i = 0; i < numEntries; i++) {
+      int32_t num = (std::rand() % (hi - lo)) - 32767;
+      inputs.push_back(num);
+      std::ostringstream os;
+      os << "insert into foo values(" << i << ", " << num << ");";
+      TestingSQLUtil::ExecuteSQLQuery(os.str());
+    }
+    EXPECT_EQ(i, numEntries);
+    txn_manager.CommitTransaction(txn);
+
+    // Fetch values from the table
+    std::vector<ResultValue> result;
+    std::vector<FieldInfo> tuple_descriptor;
+    std::string error_message;
+    int rows_affected;
+    std::string sql_line = "select id, abs(income) from foo;";
+
+    TestingSQLUtil::ExecuteSQLQuery(sql_line, result, tuple_descriptor,
+                                    rows_affected, error_message);
+    for (i = 0; i < numEntries; i++) {
+      std::string result_id(
+              TestingSQLUtil::GetResultValueAsString(result, (2 * i)));
+      std::string result_income(
+              TestingSQLUtil::GetResultValueAsString(result, (2 * i) + 1));
+      int id = std::stoi(result_id);
+      int income = std::stoi(result_income);
+      EXPECT_EQ(id, i);
+      EXPECT_EQ(income, abs(inputs[i]));
```

Comment:
Can you use [`TestingSQLUtil::ExecuteSQLQueryAndCheckResult`](https://github.com/cmu-db/peloton/blob/master/test/include/sql/testing_sql_util.h#L81)? I find it to be much clearer to see what the test is doing.